### ISSUE TITLE
Trouble with event object

### ISSUE THREAD

**Issue by feus4177** [socketio/socket.io-client-cpp#9]
I can't seem to access the public interface of the event object. So far I seem to be able to get everything else to work, I can send and receive messages from the server, but I have to use the `[&](string const& name, sio::message::ptr const& data, bool is_ack, sio::message::ptr& ack_resp)` signature. I'd like to be able to use the `[&](sio::event& ev)` signature but when I try to do something like `ev.get_messge()` my complier throws an error
```
Undefined symbols for architecture x86_64:
  "sio::event::get_message() const", referenced from:
      _main in test.cpp.o
```

**Comment by melode11** [comment]
Checked the code, the `get_message` definition is right there in `socket.cpp` file.
Can you show me more detail about the version you used and the development environment?

**Comment by melode11** [comment]
This is my mistake of using inline functions, they work when integrate with source code, but won't work when integrate with static lib.
Fix here 035ad01cc08096ebe3e8adffce2ecbc2bdfbcfa3

**Comment by feus4177** [comment]
Thanks! C++ isn't my forte, I just know enough to get myself in trouble.
